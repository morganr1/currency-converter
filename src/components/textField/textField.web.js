import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import FormControl from '@material-ui/core/FormControl';

const useStyles = makeStyles(theme => ({
    textField: {
        marginLeft: theme.spacing(2),
        marginRight: theme.spacing(2),
        width: 400,
    },
}));

export const TextFieldComponent = ({ value, onChangeCallback, label }) => {
    const classes = useStyles();
    return (
        <FormControl variant="outlined">
            <TextField
                aria-label={`${label} text field`}
                autoFocus={true}
                className={classes.textField}
                id={`${label}-input`}
                label={label}
                value={value}
                onChange={onChangeCallback}
                margin="normal"
                variant="outlined"
                type="number"
                required
                InputProps={{
                    inputMode: "numeric",
                    inputProps: {
                        min: 1
                    }
                }}
            />
        </FormControl>
    );
};
