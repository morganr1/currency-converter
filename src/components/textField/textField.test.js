import React from 'react';
import TextField from '@material-ui/core/TextField';
import { TextFieldComponent } from './textField.web';
import { shallow } from 'enzyme';

describe('TextField Component', () => {
  const mockProps = {
    value: 1,
    onChangeCallback: jest.fn(),
    label: 'test label'
  };
  const wrapper = shallow(<TextFieldComponent {...mockProps} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('change event calls the onChangeCallback', () => {
    const textField = wrapper.find(TextField);
    expect(textField).toBeDefined();
    textField.simulate('change', { target: { value: 2 } });
    expect(mockProps.onChangeCallback).toHaveBeenCalledWith({
      target: { value: 2 }
    });
  });
});
