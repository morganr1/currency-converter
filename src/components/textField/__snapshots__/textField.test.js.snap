// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component renders correctly 1`] = `
<WithStyles(ForwardRef(FormControl))
  variant="outlined"
>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "inputMode": "numeric",
        "inputProps": Object {
          "min": 1,
        },
      }
    }
    aria-label="test label text field"
    autoFocus={true}
    className="makeStyles-textField-1"
    id="test label-input"
    label="test label"
    margin="normal"
    onChange={[MockFunction]}
    required={true}
    type="number"
    value={1}
    variant="outlined"
  />
</WithStyles(ForwardRef(FormControl))>
`;
